library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java

```diff
@@ -23,13 +23,17 @@
 
     public boolean canDownload(DownloadBatch downloadBatch) {
         if (isDownloadManagerReadyToDownload(downloadBatch)) {
-            Download download = downloadMarshaller.marshall(downloadBatch);
-            return downloadClientReadyChecker.isAllowedToDownload(download);
+            clientAllowToDownload(downloadBatch);
         }
 
         return false;
     }
 
+    boolean clientAllowToDownload(DownloadBatch downloadBatch) {
```

Comment:
The client is always allowed :P Is the download that might not, what about clientAllowsToDownload(...)